src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,48 @@
+package roomescape.service;
+
+import static roomescape.exception.ExceptionMessage.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import roomescape.domain.Reservation;
+import roomescape.dto.request.ReservationRequest;
+import roomescape.dto.response.ReservationResponse;
+import roomescape.exception.BaseException;
+
+@Service
+public class ReservationService {
+
+    private final List<Reservation> reservations = new ArrayList<>();
+    private final AtomicLong index = new AtomicLong(0);
+
+    public List<ReservationResponse> getReservations() {
+        return reservations.stream()
+            .map(ReservationResponse::toDto)
+            .toList();
+    }
+
+    public ReservationResponse createReservation(ReservationRequest reservationRequest) {
+        Reservation reservation = Reservation.toDomain(index.incrementAndGet(),reservationRequest);
+        reservations.add(reservation);
+        return ReservationResponse.toDto(reservation);
+    }
+
+    public void deleteReservation(Long id) {
+        Reservation deleteReservation = null;
+        for (Reservation reservation : reservations) {
+            if (reservation.getId().equals(id)) {
+                deleteReservation = reservation;
+            }
+        }
+        deleteReservation = Optional.ofNullable(deleteReservation)
+            .orElseThrow(()->new BaseException(NOT_EXIST_RESERVATION));
```

Comment:
해당 코드는 다음과 같이 표현할 수도 있을 것 같아요~

```suggestion
Reservation deleteReservation = reservations.stream()
  .filter(reservation -> reservation.getId() == id)
  .findFirst() // 혹은 findAny()
  .orElseThrow(() -> new BaseException(NOT_EXIST_RESERVATION));
```

Reply:
stream을 사용하니 코드가 더 예쁘고 간결해지네요..!
stream을 이용한 코드로 수정하겠습니다!